src/Protobuf.System.Text.Json/FieldTypeResolver.cs

```diff
@@ -32,6 +33,8 @@ public static Type ResolverFieldType(FieldDescriptor fieldDescriptor, Dictionary
                 return typeof(bool);
             case FieldType.String:
                 return typeof(string);
+            case FieldType.Bytes:
+                return typeof(ByteString);
             case FieldType.Message when fieldDescriptor.MessageType.ClrType is { } clrType:
```

Comment:
Should we add a `ByteString` condition under the `FieldType.Message` case too?

Reply:
If we implement `ByteStringConverter` as `JsonConverter<ByteString?>` instead of `InternalConverter` it should work for all the cases. 

You could extend `MessageWithMaps` and corresponding `MessageWithMapsTests` with additional property:
```
map<string, bytes> map_string_to_bytes_type = 3;
```
and give it a try.